File: LeetCode/C#/1773. Count Items Matching a Rule.cs
public class Solution {
    public int CountMatches(IList<IList<string>> items, string ruleKey, string ruleValue) {
        var numOfMatches = 0;
        var indexToSearch = 0;
        
        if(ruleKey == "color")
        {
            indexToSearch = 1;
        }
        else if(ruleKey == "name")
        {
            indexToSearch = 2;
        }
        
        for(var i = 0; i < items.Count(); i++)
        {
            if(string.Equals(items[i][indexToSearch], ruleValue))
            {
                numOfMatches++;
            }
        }
        
        return numOfMatches;
    }
}

File: LeetCode/C++/1773. Count Items Matching a Rule.cpp
class Solution
{
public:
    int countMatches(vector<vector<string>> &items, string ruleKey, string ruleValue)
    {

        int count = 0;

        int indexToCheck = 0;
        if (ruleKey == "color")
        {
            indexToCheck = 1;
        }
        else if (ruleKey == "name")
        {
            indexToCheck = 2;
        }

        for (int i = 0; i < items.size(); i++)
        {
            if (items[i][indexToCheck] == ruleValue)
            {
                count++;
            }
        }

        return count;
    }
};